### ISSUE TITLE
Attribute assignment with reserved keywords produces no error

### ISSUE THREAD

**Issue by Jaakkonen** [microsoft/pyright#2122]
**Describe the bug**
Assignment to reserved keyword attributes produces no error.
```python
# file.py
class A:
  def __init__(self):
    self.from = 1
```

**To Reproduce**
```sh
$ pyright file.py
0 errors, 0 warnings, 0 infos
Completed in 0.557sec
```

**Expected behavior**
```sh
$ pyright file.py
file.py:4:6 - error: Cannot use reserved keyword as attribute name
```

Behavior with mypy:
```sh
$ mypy file.py
file.py:4: error: invalid syntax
Found 1 error in 1 file (errors prevented further checking)
```
Behavior when executing:
```sh
$ python3 file.py
  File "$PWD/file.py", line 4
    self.from = 1
         ^

SyntaxError: invalid syntax
```
**VS Code extension or command-line**
No error is produced with VS Code 1.1.157 or with CLI version 1.1.158.

**Comment by erictraut** [comment]
Thanks for the suggestion. Pyright's parser was too permissive in allowing keywords to be used as identifiers. I've updated it to be more restrictive, matching the behavior of the Python interpreter.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
